contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -146,89 +85,48 @@ contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockabl
         }
 
         uint256 _withdrawShares = withdrawShares;
-        return
-            _withdrawShares == 0
-                ? precision
-                : withdrawPool.mul(precision).div(_withdrawShares);
+        return _withdrawShares == 0 ? precision : withdrawPool.mul(precision).div(_withdrawShares);
     }
 
-    function buyVoucher(uint256 _amount, uint256 _minSharesToMint) public {
-        _buyVoucher(_amount, _minSharesToMint);
+    function getLiquidRewards(address user) public view returns (uint256) {
+        return _calculateReward(user, getRewardPerShare());
     }
 
-    function _buyVoucher(uint256 _amount, uint256 _minSharesToMint) internal returns(uint256) {
-        _withdrawAndTransferReward(msg.sender);
-        uint256 amountToDeposit = _buyShares(_amount, _minSharesToMint, msg.sender);
-        require(stakeManager.delegationDeposit(validatorId, amountToDeposit, msg.sender), "deposit failed");
-        return amountToDeposit;
+    function getRewardPerShare() public view returns (uint256) {
+        return _calculateRewardPerShareWithRewards(stakeManager.delegatorsReward(validatorId));
     }
 
-    function _getTotalStake(address user) internal view returns(uint256, uint256) {
-        uint256 shares = balanceOf(user);
-        uint256 rate = exchangeRate();
-        if (shares == 0) {
-            return (0, rate);
-        }
+    /**
+        Public Methods
+     */
 
-        return (rate.mul(shares).div(_getRatePrecision()), rate);
+    function buyVoucher(uint256 _amount, uint256 _minSharesToMint) public {
```

Comment:
It's good to have `amountToDeposit` as return value to `buyVoucher`. Other contracts can use that value.

Reply:
Done.